File: SDKs/ideaconsult.net/enanomapper/4.0.0/python/src/sdk/models/operations/getsubstancestructures.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from sdk.models import shared


@dataclass
class GetSubstanceStructuresPathParams:
    db: shared.AmbitDatabaseIDEnum = field(metadata={'path_param': { 'field_name': 'db', 'style': 'simple', 'explode': False }})
    uuid: str = field(metadata={'path_param': { 'field_name': 'uuid', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetSubstanceStructuresQueryParams:
    page: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'page', 'style': 'form', 'explode': True }})
    pagesize: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'pagesize', 'style': 'form', 'explode': True }})
    

@dataclass
class GetSubstanceStructuresRequest:
    path_params: GetSubstanceStructuresPathParams = field()
    query_params: GetSubstanceStructuresQueryParams = field()
    

@dataclass
class GetSubstanceStructuresResponse:
    content_type: str = field()
    status_code: int = field()
    dataset: Optional[shared.Dataset] = field(default=None)
    

File: SDKs/ideaconsult.net/nanoreg/4.0.0/go/pkg/models/operations/getsubstancestructures.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetSubstanceStructuresPathParams struct {
	Db   shared.AmbitDatabaseIDEnum `pathParam:"style=simple,explode=false,name=db"`
	UUID string                     `pathParam:"style=simple,explode=false,name=uuid"`
}

type GetSubstanceStructuresQueryParams struct {
	Page     *int64 `queryParam:"style=form,explode=true,name=page"`
	Pagesize *int64 `queryParam:"style=form,explode=true,name=pagesize"`
}

type GetSubstanceStructuresRequest struct {
	PathParams  GetSubstanceStructuresPathParams
	QueryParams GetSubstanceStructuresQueryParams
}

type GetSubstanceStructuresResponse struct {
	ContentType string
	Dataset     *shared.Dataset
	StatusCode  int64
}
